### ISSUE TITLE
question about activation function

### ISSUE THREAD

**Issue by huyufeng0407** [Beomi/InfiniTransformer#4]
`    def _retrieve_from_memory(self, query_states):
        # Retrieve context from compressive memory using linear attention (Eq. 3)
        if self.memory is None:
            debug_print("[Retrieve] No memory found")
            return torch.zeros_like(query_states)
        debug_print("[Retrieve] query_states.shape", query_states.shape)
        debug_print("[Retrieve] self.memory.shape", self.memory.shape)
        memory_output = torch.matmul(query_states, self.memory) / self.norm_term
        return memory_output

    def _update_memory(self, key_states, value_states):
        # Update compressive memory with new key-value states (Eq. 4)
        if self.memory is not None:
            self.memory = self.memory + torch.matmul(
                key_states.transpose(-2, -1), value_states
            )
            debug_print("[Update] self.memory.shape", self.memory.shape)
        else:
            self.memory = torch.matmul(key_states.transpose(-2, -1), value_states)
            debug_print("[Update] self.memory.shape", self.memory.shape)
        if self.norm_term is not None:
            self.norm_term = self.norm_term + key_states.sum(dim=-2)
        else:
            self.norm_term = key_states.sum(dim=-2)`

In the parper, 

<img width="1084" alt="infoflow 2024-04-15 16-27-52" src="https://github.com/Beomi/InfiniTransformer/assets/29395342/70bc4d2c-6c76-49ea-86c1-9d3f4d812710">

if you forget to add the activation function ?

**Comment by huyufeng0407** [comment]
the same question in memory retrieve

**Comment by Beomi** [comment]
I thought I've implemented that at early stage, but it seems I just deleted my draft with that activation function. thanks for pointing out! I just updated my code ;)
